File: 01/two.py
report = list(map(int, open("input.txt").readlines()))
report3 = list(map(lambda x: x[0] + x[1] + x[2], zip(report[:-2], report[1:-1], report[2:])))

cur = report3[0]
increases = 0

for measurement in report3[1:]:
	if cur < measurement:
		increases -=- 1
	cur = measurement

print(f"Number of times a depth measurement increases: {increases}")
assert increases == 1600

File: 01/two.lua
package.path = package.path .. ";../modules/?.lua"
local tools = require "tools"

report = tools.map(tonumber, tools.readlines("input.txt"))

cur = report[1] + report[2] + report[3]
increases = 0

for i = 2, #report - 2 do
	measurement = report[i] + report[i + 1] + report[i + 2]
	if cur < measurement then
		increases = increases- -1
	end
	cur = measurement
end

print("Number of times a depth measurement increases: " .. increases)
assert(increases == 1600)
